python/ql/lib/semmle/python/Concepts.qll

```diff
@@ -105,6 +105,72 @@ module FileSystemWriteAccess {
   }
 }
 
+/**
+ * A data-flow node that may set or unset Cross-site request forgery protection
+ * in a global manner.
+ *
+ * Extend this class to refine existing API models. If you want to model new APIs,
+ * extend `CsrfProtectionSetting::Range` instead.
+ */
+class CsrfProtectionSetting extends DataFlow::Node instanceof CsrfProtectionSetting::Range {
+  /**
+   * Gets the boolean value corresponding to if CSRF protection is enabled
+   * (`true`) or disabled (`false`) by this node.
+   */
+  boolean getVerificationSetting() { result = super.getVerificationSetting() }
```

Comment:
This was copied from Ruby, but I think a predicate would work just as well and am happy to change it. We can unify now or once we try to share.

Reply:
Ok. Then it would be good to align with ruby on this, since the 3-state of `true`/`false`/`none()` could be important for them (such as `app.setCsrfEnabled(<unknown value>)`) -- maybe that IS actually a good thing to be able to model from the get go. I think it is, so ignore this comment :heavy_check_mark: